rubicon/repository/base.py

```diff
@@ -38,12 +38,29 @@ def __init__(self, root_dir, **storage_options):
 
     def _ls_directories_only(self, path):
         """Returns the names of all the directories at path `path`."""
-        return [
+        directories = [
             os.path.join(p.get("name"), "metadata.json")
             for p in self.filesystem.ls(path, detail=True)
             if p.get("type", p.get("StorageClass")).lower() == "directory"
         ]
 
+        # TODO - discuss changing this to ignore directories that have dotfiles in their paths
```

Comment:
@ryanSoley pointed this out -- leaving this up for discussion. There are pros and cons to both. One pro I like of relying on `_cat_paths` is that it's extensible to other errors we may encounter in the future rather than only focusing on the dotfile problem

Reply:
I'm cool with the `_cat_paths` function. that makes sense to me. it may be useful to throw a warning when it hits an error in there